pom/metadata.go

```diff
@@ -21,13 +21,17 @@ type Metadata struct {
 }
 
 /* For reading Metadata from downloaded XML file. */
-func MetadataFromReader(reader io.ReadCloser) (*Metadata, error) {
-	defer reader.Close() // TODO unhandled error
+func MetadataFromReader(reader io.ReadCloser) (meta *Metadata, err error) {
+	defer func() {
+		cerr := reader.Close()
+		if err == nil {
+			err = cerr
+		}
+	}()
```

Comment:
I know you probably didn't enjoy adding this code, as it looks dirty, but Go doesn't have a good pattern for handling errors caused in or by deferred functions

Reply:
Yeah, it's ugly as sin, bu what can you do. Can't wait for `1.15`.